pkgs/by-name/mo/monaspace/package.nix

```
@@ -58,4 +33,35 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     maintainers = [ ];
     platforms = lib.platforms.all;
   };
+  makeFont =
+    font:
+    stdenvNoCC.mkDerivation {
+      inherit (font) pname;
+      inherit (fonts) version;
+      src = fetchzip {
+        url = "${baseUrl}/v${fonts.version}/${font.pname}-v${fonts.version}.zip";
+        stripRoot = false;
+        inherit (font) hash;
+      };
+
+      installPhase = ''
+        runHook preInstall
+
+        # The result of unzipping the file is a folder containing a folder containing the font file/-s
+        install -Dm644 **/**/*.* -t $out/share/fonts/${font.destination}
+
+        runHook postInstall
+      '';
+      inherit meta;
+    };
+  makePackages = lib.pipe fonts.fonts [
+    (builtins.map (font: {
+      inherit (font) name;
+      value = makeFont font;
+    }))
+    lib.listToAttrs
+  ];
+in
+makePackages.static.overrideAttrs (old: {
+  passthru = makePackages;
 })
```

`overrideAttrs` should not be used in upstream `nixpkgs`, see [the contribting guidance for the reason why](https://github.com/NixOS/nixpkgs/tree/master/pkgs#overrideattrs-and-overridepythonattrs).

I was not aware that `pkgs/by-name` enforced a requirement that the output of `pkgs.${name}` be a single `mkDerivation` [noted here](https://github.com/NixOS/nixpkgs-vet/blob/main/README.md#nix-evaluation-checks), I think that this is why `makePackages` likely was flagged in the CI as it is a set of derivations.

We can use `symlinkJoin` instead. In this implementation the default behaviour is that `pkgs.monaspace` installs the `frozen` `static` and `variable` fonts, `pkgs.monaspace.woff` installs the webfont variants and individual font variants are available as `pkgs.monaspace.variant`, for example `pkgs.monaspace.frozen`. I have set it up in this way as this is the behavior the package had before.
```suggestion
symlinkJoin {
  pname = "monaspace";
  inherit (fonts) version;
  paths = builtins.attrValues noWoffFonts;
  passthru = allFonts // {
    woff = woffFonts;
  };
  inherit meta;
}
```